### ISSUE TITLE
Monorepo setup issue

### ISSUE THREAD

**Issue by purplemana** [maty21/mocha-sidebar#152]
### Description
I have a yarn workspaces based mono-repo. The code is located at https://github.com/purplemana/alchemist. It's a TypeScript project. There are multiple modules under `packages`. Every module has its own `tsconfig.json`. Specs are co-located with the source files with *.test.ts extension. The most complex module is `api-server`, where I have defined `paths` in `tsconfig.json` to resolve synthetic `import`s. To make the compiled JS code run, I use `tsconfig-paths` module with mocha, e.g. 

    "test": "yarn build && cross-env NODE_ENV=test mocha -r tsconfig-paths/register ./build/**/*.test.js"

For equivalent configuration, I have tried to configure the extension with these settings:

    mocha path: mocha/lib/mocha.js
    test files location: packages/*/build/**/*.test.js
    files to ignore: **/.git/**/*,**/node_modules/**/*
    environmets: {"NODE_ENV":"test"}
    requires: ["tsconfig-paths/register"]
    options:  {}

When I run the tests from the extension, I get this error -

    Error: Cannot find module 'fixtures/db'
        at Function.Module._resolveFilename (module.js:547:15)
        at Function.Module._load (module.js:474:25)
        at Module.require (module.js:596:17)
        at require (internal/module.js:11:18)
        at Object.<anonymous> (c:\code\alchemist\packages\api-server\build\src\core-api\auth\createUser.test.js:23:14)

My guess is that this is because mocha is not being run relative to each module's tsconfig.json location, e.g. `packages/api-server`. This is causing the `tsconfig-paths` module to not be able to resolve the synthetic paths.

How can I make this extension work for my setup?
Thanks

### Versions

- VScode: v1.28.0
- Mocha SideBar: v0.20.18 
- Node Version: v10.6.0

### OS version

- [ ] Linux 
- [ ] Mac
- [x] Windows
- [x] Windows Subsystem for Linux running Ubuntu

**Comment by maty21** [comment]
I'll try to check it for you guys

**Comment by klis87** [comment]
@maty21 btw, for some reason `chai/register-assert` is not found during registration, so question, is this monorepo too, or this plugin is designed in such a way that it uses own node_modules for dependencies like Babel during starting `mocha`? Notice for instance that I use Babel 7 and this plugin example suggests using older Babel which doesnt use namespace @babel yet. If thats a separate issue I could create one, unless it has sth to do with monorepo as well. Anyway, thx for awesome plugin, if those issues are solved, I will get even bigger motivation to write tests ;)

**Comment by maty21** [comment]
I don't think that that is your problem imo it's probably realted to a
incorrect setting . What is the mocha cli command you running and working
for you

**Comment by klis87** [comment]
@maty21 with those settings:
```js
  "mocha.files.glob": "src/**/*.spec.js",
  "mocha.requires": [
    "@babel/register",
    "chai/register-assert"
  ],
  "mocha.env": {
    "BABEL_ENV": "test"
  },
  "mocha.coverage": {
    "enable": false,
    "decoration": false,
    "runWithInterval": false,
    "autoUpdateInterval": 20000,
    "activeOnStart": false,
    "runAfterTest": false,
    "runCoverageAfterFileSave": false,
    "reporters": []
  },
  "mocha.subdirectory": "/home/klis87/projects/redux-saga-requests/packages/redux-saga-requests"
```
I can see in `output`:
```
test runs with those args:

  files: [],
  options: {},
  grep:null,
  requires: ["@babel/register","chai/register-assert"],
  rootpath: "/home/klis87/projects/redux-saga-requests/packages/redux-saga-requests"
```

So for some reason even my tests are not found, when I try to update mocha path, it doesnt work at all, if I try to mess with mocha path too then it doesnt work at all, for example this `command mocha-maty.autoPlay not found`

**Comment by klis87** [comment]
and for coverage commend: 
```
coverage running the following command /home/klis87/.vscode/extensions/maty.vscode-mocha-sidebar-0.20.20/node_modules/nyc/bin/nyc.js --all  --reporter=html --reporter=text --reporter=lcov  /home/klis87/.vscode/extensions/maty.vscode-mocha-sidebar-0.20.20/node_modules/mocha/bin/mocha --require @babel/register --require chai/register-assert  src/**/*.spec.js --exit
```

**Comment by klis87** [comment]
Maybe if thats help too, I use following babel config:
```js
module.exports = {
  presets: [
    [
      '@babel/preset-env',
      {
        loose: true,
        modules: process.env.BABEL_ENV === 'es' ? false : 'commonjs',
      },
    ],
  ],
  plugins: [
    process.env.BABEL_ENV === 'coverage' && 'istanbul',
    ['@babel/plugin-proposal-object-rest-spread', { loose: true }],
    [
      '@babel/plugin-transform-runtime',
      {
        corejs: false,
        helpers: true,
        regenerator: true,
      },
    ],
  ].filter(Boolean),
};
```

**Comment by klis87** [comment]
generally I think this plugin should NEVER use its own version of mocha etc, it should work only if user has those dependencies installed for a project. This is like eslint plugin works for instance. If someone needs plugin for mocha, one will have this installed already, so there is no use for plugins's mocha version as fallback imho, as when someone doesnt have this installed, it means a given project doesnt use mocha and this plugin shouldn't even be active.

**Comment by klis87** [comment]
@maty21 I guess I misunderstood your question, apologies. for testing my command in package.json in a package dir I use: `"test": "cross-env BABEL_ENV=test mocha -r @babel/register -r chai/register-assert 'src/**/*.spec.js'"`

**Comment by maty21** [comment]
Tks man if you have a sample of project that not working it will be nice so
I will able to debug it for you

**Comment by klis87** [comment]
Ok, I am not sure I will be able to do this today, if not over the weekend, I will try to your your plugin in simpler scenarios, adding more and more complex and then I will give you the link for the repo and point what is problematic. Thanks!

**Comment by klis87** [comment]
or you can check it wright here - https://github.com/klis87/redux-saga-requests, it is open source

**Comment by klis87** [comment]
@maty21 here you can see example - https://github.com/klis87/mocha-sidebar-example this is extremely simple one for now, but even this seems not working. For some reason it sometimes works sometimes it doesn't on project reload (just nothing is decorated or shown in sidebar). Also command `run all tests` cannot find any test to run. I you prefer, I can create a separate issue for this. Once this is fixed, I will try to add more stuff until we get monorepo working.

I use newest VsCode btw.
